product_supplierinfo_for_customer/models/product_product.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2015 OdooMRP team
+# Copyright 2015 AvanzOSC
+# Copyright 2015 Tecnativa
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import api, models
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    @api.multi
+    def _get_price_from_supplierinfo(self, partner_id):
+        if partner_id:
+            for product in self:
+                supplierinfo = self.env['product.supplierinfo'].search(
+                    ['|', ('product_tmpl_id', '=', product.product_tmpl_id.id),
+                     ('product_id', '=', product.id),
+                     ('type', '=', 'customer'),
+                     ('name', '=', partner_id)])
+                if supplierinfo:
+                    return supplierinfo.price
+        return 0.0
+
+    @api.multi
+    def price_compute(
+            self, price_type, uom=False, currency=False, company=False):
+        if price_type == 'partner':
+            partner = self.env.context.get('partner_id', False) or \
+                self.env.context.get('partner', False)
+            for product in self:
+                price = product._get_price_from_supplierinfo(partner)
+                if not price:
+                    return super(ProductProduct, self).price_compute(
```

Comment:
Returning from inside the loop doesn't seem to be a good idea unless you want to return the _first you find_ like in `_get_price_from_supplierinfo`.
I suggest to get the prices from the `super(ProductProduct, self).price_compute(price_type, uom, currency, company)`; then overwrite the result with what you find for the supplier
and then overwrite the entries for which `_get_price_from_supplierinfo` returns something different from `0.0`.